src/transformers/models/mgp_str/modeling_mgp_str.py

```diff
@@ -0,0 +1,363 @@
+        """Initialize the weights"""
+        if isinstance(module, (nn.Linear, nn.Conv2d)):
+            module.weight.data = nn.init.trunc_normal_(module.weight.data, mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module: MGPSTREncoder, value: bool = False) -> None:
+        if isinstance(module, MGPSTREncoder):
+            module.gradient_checkpointing = value
+
+
+MGP_STR_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`MGPSTRConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+
+@add_start_docstrings(
+    "The MGP-STR Model transformer with classification heads on top.",
+    MGP_STR_START_DOCSTRING,
+)
+class MGPSTRModel(MGPSTRPreTrainedModel):
+    def __init__(self, config: MGPSTRConfig):
+        super().__init__(config)
+        self.config = config
+        self.num_tokens = 2 if config.distilled else 1
+
+        self.cls_token = nn.Parameter(torch.zeros(1, 1, config.embed_dim))
+        self.patch_embed = MGPSTRPatchEmbeddings(config.img_size, config.patch_size, config.in_chans, config.embed_dim)
+        num_patches = self.patch_embed.num_patches
+        self.pos_embed = nn.Parameter(torch.zeros(1, num_patches + self.num_tokens, config.embed_dim))
+        self.pos_drop = nn.Dropout(p=config.drop_rate)
+
+        self.encoder = MGPSTREncoder(
+            config.embed_dim,
+            config.depth,
+            config.num_heads,
+            config.mlp_ratio,
+            config.qkv_bias,
+            config.drop_rate,
+            config.attn_drop_rate,
+            config.drop_path_rate,
+            config.norm_layer,
+            config.act_layer,
+        )
+
+        self.char_a3_module = MGPSTRA3Module(config.embed_dim, config.max_token_length)
+        self.bpe_a3_module = MGPSTRA3Module(config.embed_dim, config.max_token_length)
+        self.wp_a3_module = MGPSTRA3Module(config.embed_dim, config.max_token_length)
+
+        self.char_head = nn.Linear(config.embed_dim, config.char_num_classes)
+        self.bpe_head = nn.Linear(config.embed_dim, config.bpe_num_classes)
+        self.wp_head = nn.Linear(config.embed_dim, config.wp_num_classes)
+
+    def forward(self, pixel_values):
```

Comment:
I suggest looking at the other modeling files to make sure all of the common arguments are included e.g. [here for CLIP](https://github.com/huggingface/transformers/blob/0db5d911fc94604f9568b4b212e005ec4600d157/src/transformers/models/clip/modeling_clip.py#L1277). All the model's `forward` methods should include `output_attentions`, `output_hidden_states` and `return_dict`. 

The output should also be a dataclass, if `return_dict=False` e.g. [CLIPVisionModelOutput](https://github.com/huggingface/transformers/blob/0db5d911fc94604f9568b4b212e005ec4600d157/src/transformers/models/clip/modeling_clip.py#L1319)